tests/lib/prepare-restore.sh

```diff
@@ -482,6 +482,13 @@ prepare_suite_each() {
             ausearch -m AVC --checkpoint "$RUNTIME_STATE_PATH/audit-stamp" || true
             ;;
     esac
+
+    find "$SNAP_MOUNT_DIR" "$LIBEXECDIR" "$MEDIA_DIR" "/var/snap/" -printf "%M %k %u:%g %p\n" > "$RUNTIME_STATE_PATH/invar.new"
```

Comment:
Advice from my iteration: use a directory for invariants. I used a .d directory with scripts that measure each aspect (packages, apparmor profiles, mount points, systemd services and more). Then a diff across those is more useful than a diff over one long file.

Reply:
IIRC some tests fiddle with the host fs outside of those paths. Crazy idea, but I wonder if using something like tripwire would help.

Reply:
are there  other relevant/interesting  part of the tree?   I can think  of /etc  and /var/*  generally?

Reply:
I see some tests replacing/fiddling with files under `/usr/{bin,lib,share/dbus-1,share/xdg-desktop-portal,src}`